<?xml version="1.0" encoding="UTF-8"?>
<class xmlns="http://xml.phpdox.net/src" full="Application\Modules\Rest\Exceptions\ToManyRequestsException" namespace="Application\Modules\Rest\Exceptions" name="ToManyRequestsException" abstract="false" final="false" start="19" end="48">
  <file path="/var/www/backend.local/Application/Modules/Rest/Exceptions" file="ToManyRequestsException.php" realpath="/var/www/backend.local/Application/Modules/Rest/Exceptions/ToManyRequestsException.php" size="1417" time="2015-07-22T21:33:28+00:00" unixtime="1437600808" sha1="0ff2df834b408c8003a08b28cba04b9ab0ccd506" relative="Modules/Rest/Exceptions/ToManyRequestsException.php"/>
  <docblock>
    <description compact="Class ToManyRequestsException. Represents an HTTP 429 error.">The 429 status code indicates that the user has sent too many
requests in a given amount of time ("rate limiting").
The response representations SHOULD include details explaining the
condition, and MAY include a Retry-After header indicating how long
to wait before making a new request...</description>
    <package value="Application\Modules\Rest"/>
    <subpackage value="Exceptions"/>
    <since value="PHP &gt;=5.6"/>
    <version value="1.0"/>
    <author value="Stanislav WEB | Lugansk &lt;stanisov@gmail.com&gt;"/>
    <filesource value="/Application/Modules/Rest/Exceptions/ToManyRequestsException.php"/>
  </docblock>
  <extends full="Application\Modules\Rest\Exceptions\BaseException" namespace="Application\Modules\Rest\Exceptions" name="BaseException"/>
  <constant type="string" name="MESSAGE" value="'Too Many Requests'">
    <docblock>
      <description/>
      <const value="HTTP response message"/>
    </docblock>
  </constant>
  <constant type="integer" name="CODE" value="429">
    <docblock>
      <description/>
      <const value="HTTP response code"/>
    </docblock>
  </constant>
  <constructor name="__construct" start="38" end="47" abstract="false" final="false" static="false" visibility="public">
    <docblock>
      <description compact="Constructor"/>
      <param description="additional info" type="array" variable="$data"/>
      <param description="If no message is given 'Too Many Requests' will be the message" type="string" variable="$message"/>
      <param description="Status code, defaults to 429" type="int" variable="$code"/>
    </docblock>
    <parameter name="data" byreference="false" type="array" default=""/>
    <parameter name="message" byreference="false" type="{unknown}" default="" constant="null"/>
    <parameter name="code" byreference="false" type="{unknown}" default="" constant="null"/>
  </constructor>
  <parent full="Application\Modules\Rest\Exceptions\BaseException" namespace="Application\Modules\Rest\Exceptions" name="BaseException">
    <extends full="Exception" namespace="" name="Exception"/>
    <constructor name="__construct" start="23" end="30" abstract="false" final="false" static="false" visibility="public">
      <docblock>
        <description compact="Constructor"/>
        <param description="If no message is given default from child" type="string" variable="$message"/>
        <param description="Status code, default from child" type="int" variable="$code"/>
        <param description="Additional data to be represent by exception throw" type="array" variable="$data"/>
      </docblock>
      <parameter name="message" byreference="false" type="{unknown}" default="" constant="null"/>
      <parameter name="code" byreference="false" type="{unknown}" default="" constant="null"/>
      <parameter name="data" byreference="false" type="array"/>
    </constructor>
  </parent>
  <parent full="Exception" namespace="" name="Exception">
    <member name="message" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The exception message"/>
      <var type="string"/>
    </docblock>
  </member>
    <member name="code" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The exception code"/>
      <var type="int"/>
    </docblock>
  </member>
    <member name="file" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The filename where the exception was created"/>
      <var type="string"/>
    </docblock>
  </member>
    <member name="line" static="false" visibility="protected">
    <default>null</default>
    <docblock>
      <description compact="The line where the exception was created"/>
      <var type="int"/>
    </docblock>
  </member>
    <constructor name="__construct" abstract="false" static="false" visibility="public" final="false">
    <docblock>
      <description compact="Construct the exception"/>
      <return type="void"/>
    </docblock>
    <parameter name="message" optional="true" byreference="false" type="string"/>
    <parameter name="code" optional="true" byreference="false" type="int"/>
    <parameter name="previous" optional="true" byreference="false" type="object" class="Exception"/>
  </constructor>
    <method name="getMessage" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the Exception message"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="getPrevious" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Returns previous Exception"/>
      <return type="Exception"/>
    </docblock>
  </method>
    <method name="getCode" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the Exception code"/>
      <return type="mixed"/>
    </docblock>
  </method>
    <method name="getFile" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the file in which the exception occurred"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="getLine" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the line in which the exception occurred"/>
      <return type="int"/>
    </docblock>
  </method>
    <method name="getTrace" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the stack trace"/>
      <return type="array"/>
    </docblock>
  </method>
    <method name="getTraceAsString" abstract="false" static="false" visibility="public" final="true">
    <docblock>
      <description compact="Gets the stack trace as a string"/>
      <return type="string"/>
    </docblock>
  </method>
    <method name="__toString" abstract="false" static="false" visibility="public" final="false">
    <docblock>
      <description compact="String representation of the exception"/>
      <return type="string"/>
    </docblock>
  </method>
  </parent>
</class>
